tests/cocotb/test_snitch_cc.py

```diff
@@ -0,0 +1,125 @@
+async def snitch_cc_dut(dut):
+    # Initialize clock
+    clock = Clock(dut.clk_i, 10, units="ns")
+    cocotb.start_soon(clock.start())
+
+    # Reset or intial values
+    dut.rst_ni.value = 0
+
+    # Wait 1 cycle to reset
+    await RisingEdge(dut.clk_i)
+
+    # Deassert reset
+    dut.rst_ni.value = 1
+
+    for i in range(CLOCK_CYCLES):
+        # Simple check if instructions are running normally
+        # instruction_value = hex(dut.i_snitch_cc.i_snitch.inst_data_i.value)
+        # instruction_address = int(dut.instruction_addr_offset.value)
+
+        # cocotb.log.info('---------- Instruction Info ----------')
+        # cocotb.log.info(f'Instruction Value: {instruction_value}')
+        # cocotb.log.info(f'Instruction Addr: {instruction_address}')
+
+        await RisingEdge(dut.clk_i)
+
+
+# Main test run
+@pytest.mark.parametrize("parameters", [{"AddrWidth": str(48), "DataWidth": str(64)}])
+def test_snitch_cc(parameters, simulator):
+    # Working paths
+    repo_path = "something"
+    tests_path = repo_path + "/tests/cocotb/"
+
+    filelist = subprocess.run(["bender", "script", "verilator"], stdout=subprocess.PIPE)
+
+    filelist = filelist.stdout.decode("utf-8").strip().split("\n")
+
+    include_folders = []
+    rtl_sources = []
+
+    for item in filelist:
+        if item == "":
+            pass
+        elif (
+            item[0] == "#"
+            or item[0] == ""
+            or item[0] == "\n"
+            or item[0:8] == "+define+"
+        ):
+            pass
+        elif item[0:8] == "+incdir+":
+            include_folders.append(item[8:].strip())
+        else:
+            rtl_sources.append(item.strip())
+
+    # Append test bench to rtl list
+    rtl_sources.append("tests/tb/tb_snitch_cc.sv")
+
+    # Specify top-level module
+    toplevel = "tb_snitch_cc"
+
+    # Specify python test name that contains the @cocotb.test.
+    # Usually the name of this test.
+    module = "test_snitch_cc"
+
+    # Specify build directory
+    sim_build = tests_path + "/sim_build/{}/".format(toplevel)
+
+    if simulator == "verilator":
+        compile_args = [
+            "-Wno-LITENDIAN",
+            "-Wno-WIDTH",
+            "-Wno-CASEINCOMPLETE",
+            "-Wno-BLKANDNBLK",
+            "-Wno-CMPCONST",
+            "-Wno-WIDTHCONCAT",
+            "-Wno-UNSIGNED",
+            "-Wno-UNOPTFLAT",
+            "-Wno-TIMESCALEMOD",
+            "-Wno-fatal",
+            "--no-timing",
+        ]
+
+        run(
+            verilog_sources=rtl_sources,
+            includes=include_folders,
+            toplevel=toplevel,
+            module=module,
+            simulator=simulator,
+            sim_build=sim_build,
+            compile_args=compile_args,
+        )
+    else:
+        run(
+            verilog_sources=rtl_sources,
+            includes=include_folders,
+            toplevel=toplevel,
+            module=module,
+            simulator=simulator,
+            timescale="1ps",
+        )
```

Comment:
```suggestion
if simulator == "verilator":
       compile_args = [ "-Wno-LITENDIAN", "-Wno-WIDTH", "-Wno-CASEINCOMPLETE", "-Wno-BLKANDNBLK", "-Wno-CMPCONST", "-Wno-WIDTHCONCAT", "-Wno-UNSIGNED", "-Wno-UNOPTFLAT", "-Wno-TIMESCALEMOD", "-Wno-fatal", "--no-timing", ]
       timescale = None
    else:
       compile_args = None
       timescale = "1ps"
    run(
        verilog_sources=rtl_sources,
        includes=include_folders,
        toplevel=toplevel,
        module=module,
        simulator=simulator,
        sim_build=sim_build,
        compile_args=compile_args,
        timescale=timescale,
    )
```    
I think we can make this shorter which can improve readability. (don't commit this as is, you should test/reformat locally first) Also, why would:
* verilator don't accept 1ps as timescale?
* verilator need a different sim_build folder?

Can you add comments explaining the reason for this?
Also, should we mention somewhere that we are only testing with modelsim besides verilator somewhere?

Reply:
Aha! Okay good questions.

| verilator don't accept 1ps as timescale?

Verilator does not measure time. It's also in its [documentation](https://itsembedded.com/dhd/verilator_1/). Quoted from there : **"Verilator is a cycle-based simulator, which means it does not evaluate time within a single clock cycle, and does not simulate exact circuit timing"**. 

| verilator need a different sim_build folder?

That was a typo, I erased it by accident when I took out some things. They need to be in the same sim_build. Thanks for catching 😄 

Just an FYI the warnings in the Verilator condition was taken from the Makefile. Particularly the `/target/common/common.mk` of the original Snitch cluster repo. Without these warning waivers, the simulator halts. Unfortunately, these are things I cannot change since they are part of the PULP repos. So they made workarounds for this which I think is a bad thing to do :( but o well... I didn't make it ^^; maybe fix later.

Thanks for the fix suggestion. Will test it out first 😃

Reply:
Just an FYI but the pre-commit fixes the `compile_args` into the per-line mode. So I guess I'll keep it like that too. IMHO I also think separating each argument per line is more readable too. But that's just me 🛩️